citest/gcp_testing/resource_snapshot.py

```diff
@@ -615,6 +699,55 @@ def __determine_added_instances(self, resource, before, after):
     else:
       return set(after[resource].response)
 
+  def delete_all_collected(
+      self, resource_type, discovery_doc, collected, bindings):
+    """Delete all the collected items.
+
+    Args:
+      resource_type: The API resource type of the items must be homogenous.
+      discovery_doc: The API discovery document.
+      collected: The API resource items as collected from the API.
+        If an aggregated_list() method was used, then this is a tuple that
+        also includes the key from the list needed by the delete() method.
+      bindings: The bindings to use for variables needed by delete. If
+        this is none, use processor's default variables.
+    """
+    api, version = discovery_doc['id'].split(':')
+    resource_container = discovery_doc.get('resources')
+    resource_segments = resource_type.split('.')
+    resource_spec = {}
+    for segment in resource_segments:
+      resource_spec = resource_container.get(segment, {})
+      resource_container = resource_spec.get('resources', {})
+
+    delete = resource_spec.get('methods', {}).get('delete', None)
+    if not delete:
+      print '*** Cannot find delete method for "{0}"'.format(resource_type)
+      return None
+
+    scope = self.__explorer.pick_scope(
+        delete.get('scopes', []), api, mutable=True)
+    agent = self.make_agent(api, version, scope,
+                            default_variables=bindings)
+
+    decorator = '[dry run] ' if self.__options.dry_run else ''
+    print '{decorator}DELETING from API={api} with scope={scope}'.format(
```

Comment:
This is the API and scope being used.
The individual elements show the type (and name and parameters)